databases/week2/class2hw.sql

```diff
@@ -0,0 +1,151 @@
+UPDATE task SET status_id = 3 WHERE id=36;
+
+--Mark a task as complete
+UPDATE task SET status_id = 3 WHERE id=35;
+
+--Delete a task
+DELETE from task where id=37;
+
+--part 2
+--Table: Class: with the columns: id, name, begins (date), ends (date)
+CREATE TABLE  class(
+id INT(10) UNSIGNED NOT NULL AUTO_INCREMENT, 
+name VARCHAR(255) NOT NULL,
+begin_date DATETIME NOT NULL,
+end_date DATETIME NOT NULL,
+PRIMARY KEY(id)) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+--Student: with the columns: id, name, email, phone, class_id (foreign key)
+CREATE Table student(
+id int(10) UNSIGNED NOT NULL AUTO_INCREMENT,
+name VARCHAR(255) NOT NULL,
+email VARCHAR(255),
+phone INT(8),
+class_id int(10) UNSIGNED NOT NULL,
+PRIMARY KEY (id), 
+CONSTRAINT `fk_class`
+FOREIGN KEY (class_id) REFERENCES class(id)ON DELETE CASCADE ON UPDATE CASCADE)
+ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+--Create an index on the name column of the student table.
+CREATE INDEX student_name ON student(name);
+
+--Add a new column to the class table named status which can only have the following values: not-started, ongoing, finished (hint: enumerations).
+ALTER TABLE class ADD status ENUM ("not-started", "ongoing", "finished");
+
+--part 3
+--Get all the tasks assigned to users whose email ends in @spotify.com
+SELECT user.id,  task.title, user.email
+FROM user, user_task, task WHERE 
+user.email LIKE "%@spotify.com" AND
+user.id=user_task.user_id AND
+user_task.task_id=task.id;
+
+-- using join (Get all the tasks assigned to users whose email ends in @spotify.com)
+SELECT * FROM user_task
+JOIN task ON task_id=task.id
+JOIN user ON user_id=user.id
+WHERE user.email LIKE '%@spotify.com';
+
+--Get all the tasks for 'Donald Duck' with status 'Not started'
+SELECT user.name as username, task.id as taskid, task.title as tasktitle,  status.name as statusname
+FROM user, user_task, task, status WHERE 
+user.name LIKE "Donald Duck" AND
+status.name LIKE "Not started" AND
+user.id=user_task.user_id AND
+user_task.task_id=task.id AND
+task.status_id=status.id;
+
+--using Join(Get all the tasks for 'Donald Duck' with status 'Not started')
+SELECT * FROM user_task 
+JOIN task ON task_id=task.id 
+JOIN user ON user_id=user.id 
+JOIN status ON status_id=status.id 
+WHERE user.name='Donald Duck' AND status.name LIKE "Not started";
+
+--Get all the tasks for 'Maryrose Meadows' that were created in september (hint: month(created)=month_number)
+
+SELECT user.name as username, task.id as taskid, task.title as tasktitle FROM user, user_task, task
+WHERE user.name LIKE "Maryrose Meadows" AND
+month(task.created)=09 AND
+user.id=user_task.user_id AND
+user_task.task_id = task.id;
+
+--using Join(Get all the tasks for 'Maryrose Meadows' that were created in september (hint: month(created)=month_number)
+SELECT * FROM user_task
+JOIN task ON task_id=task.id
+JOIN user ON user_id=user.id
+WHERE user.name='Maryrose Meadows' AND month(task.created) = 09;
```

Comment:
try this if it makes any output

```
SELECT title, month(created), user.name Created_by FROM task
JOIN user ON task.user_id=task.id
 WHERE user.name='Maryrose Meadows' AND month(task.created) = 09;
```
```

Reply:
I am getting an error, because user_id is not a column in task table. what I observed is this result is depending on 3 tables.
the relationship is id in user table = user_id in user_task table 
and
task_id in user_task table = id in task table.